### PR TITLE
[doc] Add RaptorX documentation to page

### PR DESCRIPTION
**Pull request by DimSumDev**
## Description
This adding the RaptorX documentation from community page to the Presto Documentation page

## Motivation and Context
See #21512

## Impact

N/A

## Test Plan

N/A

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:12]: internal to whom/where? Maybe just "RaptorX aims to boost..."
No comma after "cache work"

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:19]: is going to be --> is
data; a few --> data. A few

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:31]: duplicate text

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:102]: in an earlier post --> earlier

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:104]: with 3MB in length starting with offset 0 --> for the 3 MB starting at offset 0
fetch --> fetches
that has not --> that have not

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:115]: requests of the same file --> requests for the same file
"Because there is a high chance part of the file has already been cached on that particular worker." sentence fragment

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (elharo) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:206]: the original post

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for this! The blog post was not written as technical documentation, so I made several revisions for style, formatting, and tone.

**Review by tdcmeehan — COMMENTED** [review]
I would suggest not using the name RaptorX, as it's currently written it was a project name.  By now this feature is long in core, and we can just refer to this as caching over disaggregated storage.

**Comment by steveburnett** [comment]
> I would suggest not using the name RaptorX, as it's currently written it was a project name. By now this feature is long in core, and we can just refer to this as caching over disaggregated storage.

Thanks! I will wait for my existing review comments to be addressed, and will address changing "RaptorX" to "caching over disaggregated storage" in a second review pass.

File: presto-docs/src/main/sphinx/connector/raptorx.rst
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/connector/raptorx.rst:51]:
```suggestion
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
Could you please merge just the user guide with the Hive documentation?  These knobs are all specific to Hive.  We can include one paragraph specific to caching, and emphasise that when used in conjunction, they achieve a sophisticated multi-layer cache.  We don't need performance benchmarks, background, references to Raptor/RaptorX, or most of the content prior to the user guide--we simply need the knobs and what they achieve in the Hive documentation.  Do you think you could rework it in this way?

**Comment by tdcmeehan** [comment]
Please reopen once comments addressed, thank you!

**Pull request closed without merging**
